lb4 app - Fails on npm start when repository service is not chosen when creating

<!-- 🚨 STOP 🚨 STOP 🚨 STOP 🚨

HELP US HELP YOU, PLEASE
- Do a quick search to avoid duplicate issues
- Provide as much information as possible (reproduction sandbox, use case for features, etc.)
- Consider using a more suitable venue for questions such as Stack Overflow, Slack, etc.

Please fill in the *entire* template below.

-->

## Steps to reproduce

<!-- Describe how to reproduce the issue -->
1. lb4 app
2. uncheck repositories
3. npm start

## Current Behavior
Receive error on migrate.ts
```
src/migrate.ts:9:13 - error TS2339: Property 'migrateSchema' does not exist on type 'MyAppApplication'.
9   await app.migrateSchema({existingSchema});
              ~~~~~~~~~~~~~
```

<!-- Describe the observed result -->

## Expected Behavior
1. app runs successfully with no additional change.
<!-- Describe what did you expect instead, what is the desired outcome? -->

## Link to reproduction sandbox

<!-- See https://loopback.io/doc/en/contrib/Reporting-issues.html#loopback-4x-bugs  -->

## Additional information

<!--
Copy+paste the output of these two commands:
  node -e 'console.log(process.platform, process.arch, process.versions.node)'
  npm ls --prod --depth 0 | grep loopback
-->

linux x64 10.23.2
+-- @loopback/boot@3.2.0
+-- @loopback/core@2.14.0
+-- @loopback/rest@9.1.2
+-- @loopback/rest-explorer@3.0.6
+-- @loopback/service-proxy@3.0.6


## Related Issues

<!-- Did you find other bugs that looked similar? -->
Receiving same error as here: https://github.com/strongloop/loopback-next/issues/5608

_See [Reporting Issues](http://loopback.io/doc/en/contrib/Reporting-issues.html) for more tips on writing good issues_

--- a/packages/cli/generators/app/index.js
+++ b/packages/cli/generators/app/index.js
@@ -135,11 +135,15 @@ module.exports = class AppGenerator extends ProjectGenerator {
     const result = super.scaffold();
     if (this.shouldExit()) return result;
 
-    const {docker} = this.projectInfo || {};
-    if (docker) return result;
+    const {docker, repositories} = this.projectInfo || {};
+    if (!docker) {
+      this.fs.delete(this.destinationPath('Dockerfile'));
+      this.fs.delete(this.destinationPath('.dockerignore'));
+    }
+    if (!repositories) {
+      this.fs.delete(this.destinationPath('src/migrate.ts.ejs'));
+    }
 
-    this.fs.delete(this.destinationPath('Dockerfile'));
-    this.fs.delete(this.destinationPath('.dockerignore'));
     return result;
   }
 

--- a/packages/cli/generators/project/templates/package.json.ejs
+++ b/packages/cli/generators/project/templates/package.json.ejs
@@ -89,12 +89,14 @@
     "docker:build": "docker build -t <%= project.name -%> .",
     "docker:run": "docker run -p 3000:3000 -d <%= project.name -%>",
 <% } -%>
-<% if (packageManager === 'yarn') { -%>
+<% if (project.repositories && packageManager === 'yarn') { -%>
     "premigrate": "yarn run build",
-<% } else { -%>
+<% } else if (project.repositories) { -%>
     "premigrate": "npm run build",
 <% } -%>
+<% if (project.repositories) { -%>
     "migrate": "node ./dist/migrate",
+<% } -%>
 <% if (packageManager === 'yarn') { -%>
     "preopenapi-spec": "yarn run build",
 <% } else { -%>

--- a/packages/cli/generators/project/templates/package.plain.json.ejs
+++ b/packages/cli/generators/project/templates/package.plain.json.ejs
@@ -90,7 +90,9 @@
     "docker:build": "docker build -t <%= project.name -%> .",
     "docker:run": "docker run -p 3000:3000 -d <%= project.name -%>",
 <% } -%>
+<% if (project.repositories) { -%>
     "migrate": "node ./dist/migrate",
+<% } -%>
     "openapi-spec": "node ./dist/openapi-spec",
 <% if (packageManager === 'yarn') { -%>
     "prestart": "yarn run rebuild",

--- a/packages/cli/test/integration/generators/app.integration.js
+++ b/packages/cli/test/integration/generators/app.integration.js
@@ -115,6 +115,21 @@ describe('app-generator with docker disabled', () => {
   });
 });
 
+describe('app-generator with repositories disabled', () => {
+  before(() => {
+    return helpers
+      .run(generator)
+      .withOptions({repositories: false})
+      .withPrompts(props);
+  });
+  it('does not generate migration files', () => {
+    assert.noFile('src/migrate.ts');
+
+    assert.noFileContent('package.json', /"premigrate": "yarn run build/);
+    assert.noFileContent('package.json', /"migrate": "node .\/dist\/migrate/);
+  });
+});
+
 describe('app-generator with --applicationName', () => {
   before(() => {
     return helpers

